File: src-java/src/main/java/kigh/rmt/timer/TimerId.java
package kigh.rmt.timer;

import java.util.Objects;

public class TimerId {
    private final String id;

    public TimerId(String id) {
        if (id.isBlank()) {
            throw new IllegalArgumentException();
        }
        // TODO more validation
        this.id = id;
    }

    public String value() {
        return id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        TimerId timerId = (TimerId) o;
        return Objects.equals(id, timerId.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: src-kotlin/src/main/kotlin/kigh/rmt/timer/TimerId.kt
package kigh.rmt.timer

import java.util.*

class TimerId(id: String) {
    private val id: String
        get() = field

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val timerId = other as TimerId
        return id == timerId.id
    }

    override fun hashCode(): Int {
        return Objects.hash(id)
    }

    init {
        require(!id.isBlank())
        // TODO more validation
        this.id = id
    }
}